lms/djangoapps/discussion/rest_api/utils.py

```diff
@@ -0,0 +1,74 @@
+"""
+Utils for discussion API.
+"""
+
+
+import functools
+from datetime import datetime
+
+from django.core.exceptions import ValidationError
+from opaque_keys.edx.keys import CourseKey
+from pytz import UTC
+
+from lms.djangoapps.courseware.courses import get_course_by_id
+from lms.djangoapps.discussion.django_comment_client.utils import (
+    has_discussion_privileges,
+    JsonError
+)
+from openedx.core.djangoapps.django_comment_common.comment_client.thread import Thread
+
+
+def discussion_accessible(endpoint_flag):
+    """
+    View decorator (with argument) to check if discussions are accessible to user.
+    """
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            """
+            Wrapper for the view that calls the view only if the user has privileges
+            and discussions are accessible.
+            """
+            def get_course_from_context():
+                """
+                Returns course that is calculated from context from request.
+                :return:
+                """
+                if endpoint_flag == 'threads':
+                    try:
+                        course_id = request.data['course_id']
+                    except KeyError as err:
+                        raise ValidationError({"course_id": ["This field is required."]}) from err
+                else:   # for now we only have 2 places where the decorator is used.
+                    try:
+                        course_id = Thread(id=request.data['thread_id']).retrieve('course_id').get('course_id')
+                    except KeyError as err:
+                        raise ValidationError({"thread_id": ["This field is required."]}) from err
+
+                course_key = CourseKey.from_string(course_id)
+                course_from_context = get_course_by_id(course_key)
```

Comment:
what is the diff of using `get_course_by_id` vs `_get_course` from `discussion/rest-api/api.py`?  it is using `get_course_with_access`